aiida/engine/processes/calcjobs/calcjob.py

```diff
@@ -280,17 +288,60 @@ def parse(self, retrieved_temporary_folder=None):
         This is called once it's finished waiting for the calculation to be finished and the data has been retrieved.
         """
         import shutil
+        from aiida.engine import ExitCode
         from aiida.engine.daemon import execmanager
 
+        try:
+            retrieved = self.node.outputs.retrieved
+        except exceptions.NotExistent:
+            return self.exit_codes.ERROR_NO_RETRIEVED_FOLDER  # pylint: disable=no-member
+
+        scheduler = self.node.computer.get_scheduler()
+        filename_stderr = self.node.get_option('scheduler_stderr')
+        filename_stdout = self.node.get_option('scheduler_stdout')
+
+        try:
+            detailed_job_info = self.node.get_attribute('detailed_job_info')
```

Comment:
Since `CalcJobNode` has the method `get_detailed_job_info` that uses the more general `SCHEDULER_DETAILED_JOB_INFO_KEY`, shouldn't we use this method instead of `get_attribute('detailed_job_info')?